frontend/src/metabase/parameters/components/FilterApplyButton/FilterApplyButton.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useCallback, useRef } from "react";
+import _ from "underscore";
+import { CSSTransition } from "react-transition-group";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+
+import {
+  getDashboardComplete,
+  getDraftParameterValues,
+  getParameterValues,
+} from "metabase/dashboard/selectors";
+import { setParameterValues } from "metabase/dashboard/actions";
+import {
+  ApplyButton,
+  BUTTON_TRANSITION_DURATION,
+  TRANSITION_CLASSNAMES_PREFIX,
+} from "./FilterApplyButton.styled";
+
+export default function FilterApplyButton() {
+  const isAutoApplyFilters = useSelector(
+    state => getDashboardComplete(state).auto_apply_filters,
+  );
+
+  const hasUnappliedParameterValues = useSelector(state => {
+    const draftParameterValues = getDraftParameterValues(state);
+    const parameterValues = getParameterValues(state);
+    return !_.isEqual(draftParameterValues, parameterValues);
+  });
+
+  const dispatch = useDispatch();
+  const handleApplyFilters = useCallback(() => {
+    dispatch((_, getState) => {
+      const draftParameterValues = getDraftParameterValues(getState());
+      dispatch(setParameterValues(draftParameterValues));
+    });
+  }, [dispatch]);
+
+  const applyButtonRef = useRef(null);
+
+  return (
+    <CSSTransition
```

Comment:
Use `react-transition-group` to allow transitioning component when mounting and unmounting

Reply:
@ranquild Just want to make sure we're on the same page about this.
1. Is `react-transition-group` discouraged in Metabase?
2. I haven't seen any doc saying it's outdated or deprecated, if we look at their updated date
    1. [`react-transition-group`](https://www.npmjs.com/package/react-transition-group) was updated 9 months ago vs 
    2. [`react-motion`](https://www.npmjs.com/package/react-motion) was updated 6 years ago
3. TBH, if we want to stay up to date, I'd rather use another modern library e.g. [`framer-motion`](https://www.framer.com/motion), but that's outside of the scope of this project.
4. `react-motion`'s [use case](https://github.com/chenglou/react-motion#what-does-this-library-try-to-solve) is for animating components with different curves that are not only limited to CSS properties, I feel it isn't worth trying to migrate over to `react-motion` for this PR's use-case which is just to animate mounting and unmounting components.

That being said, what do you think about leaving `react-transition-group` change as is? Or at least I could remove the use of `react-transition-group` and don't animate mounting and unmounting components if we don't really like to use `react-transition-group`

Reply:
@WiNloSt would it be possible to do the transition with pure css in styled components like you did before? Are there any issues with that?

Reply:
transitioning mounting and unmounting were never support with pure CSS without a lot of work. The previous version didn't fade apply button in and out on mounting and unmounting the apply button component.

We need to remove it from the dom entirely as discussed with Maz so when `auto_apply_filters` is `true` the apply button won't take any space.

Previously when the component is mounting it just appear on the DOM, so to animate mounting the component we need a mechanism which add class name to the component one after another in a different tick so CSS could transition from say opacity 0 to 1. This is the work that's handled by `CSSTransition` from `react-transition-group`. The same goes with unmounting the component, without some work to keep the component around for the whole amount of the animation, if we just conditionally render the component, it will just be removed from the DOM without a chance to be animated, this is the work that's also handled by these kinds of transition/animation libraries.

Reply:
@WiNloSt Is it even possible that we need to animate the button when the auto_apply_filters is changed? It can only change in the edit mode (or while loading in milestone 2), there are no draft parameter values at this stage. I'd vote for a simple solution with a css transition like you did before. Would that be possible?

Reply:
@ranquild Yeah pure CSS would be better, in this case, to keep things simple. There are cases, that we (might) need to animate unmounting and mounting the component that is when you have an unapplied filters so there's the apply button visible then you toggle the auto-apply filters back to on. This should remove the apply button out of the dom. I mean it'd be nice just to keep the experience consistent when removing them from the dom and fade them out, but if that begs more question I'm cool with only using pure CSS and not animating them on mount and unmount.

So I'll animate the apply button only when it is already on the DOM, no animation when unmounting or mounting (which is rare anyway). Sounds good?

Reply:
Yep sounds great